implants/lib/eldritch/src/pivot/arp_scan_impl.rs

```diff
@@ -1,5 +1,296 @@
+                                    });
+                                    break;
+                                }
+                                println!("Failed to find {} in HashMap", ip);
+                                return;
+                            }
+                            Err(err) => {
+                                println!("Failed to get lock on data: {}", err);
+                                return;
+                            }
+                        }
+                    }
+                }
+                Err(e) => {
+                    if e.kind() == std::io::ErrorKind::TimedOut {
+                        continue;
+                    }
+                    println!("Error receiving packet: {}", e);
+                    return;
+                }
+            }
+        }
+    }
+}
+
+#[cfg(not(target_os = "windows"))]
+pub fn handle_arp_scan(
+    target_cidrs: Vec<String>,
+) -> Result<HashMap<Ipv4Addr, Option<ArpResponse>>> {
+    let listener_out: Arc<Mutex<HashMap<Ipv4Addr, Option<ArpResponse>>>> =
+        Arc::new(Mutex::new(HashMap::new()));
+    let target_cidrs = target_cidrs
+        .iter()
+        .map(|cidr| {
+            let (addr, prefix) = cidr.split_at(
+                cidr.find('/')
+                    .ok_or(anyhow::anyhow!("Failed to find / in Network {}", cidr))?,
+            );
+            let addr = match Ipv4Addr::from_str(addr) {
+                Ok(addr) => addr,
+                Err(_) => {
+                    return Err(anyhow::anyhow!("Invalid IPv4 address: {}", addr));
+                }
+            };
+            let prefix: u8 = match prefix[1..].parse() {
+                Ok(prefix) => prefix,
+                Err(_) => {
+                    return Err(anyhow::anyhow!("Invalid CIDR prefix: {}", prefix));
+                }
+            };
+            let network = match Ipv4Network::new(addr, prefix) {
+                Ok(network) => network,
+                Err(_) => {
+                    return Err(anyhow::anyhow!("Invalid CIDR: {}", cidr));
+                }
+            };
+            Ok(network)
+        })
+        .collect::<Result<Vec<Ipv4Network>>>()?;
+    for target_cidr in target_cidrs {
+        for ip in target_cidr.iter() {
+            match listener_out.lock() {
+                Ok(mut listener_lock) => {
+                    listener_lock.insert(ip, None);
+                }
+                Err(err) => return Err(anyhow::anyhow!("Failed to get lock on IP List: {}", err)),
+            }
+        }
+    }
+    let interfaces = datalink::interfaces();
+    for interface in interfaces {
+        let inner_out = listener_out.clone();
+        let inner_interface = interface.clone();
+        let thread = std::thread::spawn(move || {
+            start_listener(inner_interface, inner_out);
+        });
+        thread.join().map_err(|err| {
+            anyhow::anyhow!(
```

Comment:
How common is an error here?
Idk how it works but could see something like loopback error ing out. If this is a common error we may want to handle it and include a "interface errored" status.

This looks good though 👌

Reply:
Should never happen